### ISSUE TITLE
Is it possible to embed images in a message?

### ISSUE THREAD

**Issue by dav3-digital** [O365/python-o365#171]
Hi,
Could anyone confirm if I can embed images in a message using this library. I keep hitting a dead end in my attempts thus far.

**Comment by roycem90** [comment]
i m not sure about embedding, but you can send attachments

if the image is already uploaded somewhere, you can use html body to embed it

**Comment by dav3-digital** [comment]
Yeah... Not really the solution that I'm looking for. I found further details in a closed issue that pretty much dismisses the library being responsible for handling it. I think the larger problem that I am having is  being unable to use setBodyHTML. Maybe because I'm using account instead of connection? not really sure.

**Comment by janscas** [comment]
> I think the larger problem that I am having is being unable to use setBodyHTML

setBodyHTML is a method from the old api (<1.0). This method no longer exists.
All message bodies are html by default (unless you change it).
So, you can pass any html code to `message.body`.

**Comment by dav3-digital** [comment]
> setBodyHTML is a method from the old api (<1.0). This method no longer exists.

That explains a lot. I thought I was going crazy. Thanks heaps. Is there a way to access the json directly and get the attachment cid? If so I think I could reference the CID of the attachment in the email template.

**Comment by janscas** [comment]
What’s an attachment CID?

That’s what the attachment contains (see properties):

https://docs.microsoft.com/en-us/graph/api/resources/attachment?view=graph-rest-1.0

**Comment by dav3-digital** [comment]
It's the content ID I wanted to try and reference. I tried using just the filename but that doesn't seem to work. I was basing my understanding of this article here:

[https://www.eliostruyf.com/sending-mails-embedded-images-via-microsoft-graph/](url)

Entirely possible I'm barking up the wrong tree but I thought if I could add the attachment and them update the template with the content ID it would display as an embedded image. I tried base64 encoding the images but some of the images are too big. Not sure how else to tackle the issue at this point.
